File: java/modelo/Persona.java
package modelo;
import excepciones.DniInvalidoException;
import java.time.LocalDate;
import java.time.Period;
public abstract class Persona {
    private final String dni;
    private String nombres;
    private String apellidos;
    private LocalDate fechaNacimiento;
    private String sexo;
    public Persona(String dni, String nombres, String apellidos, LocalDate fechaNacimiento, String sexo) {
        validarDni(dni);
        this.dni = dni;
        this.nombres = nombres;
        this.apellidos = apellidos;
        this.fechaNacimiento = fechaNacimiento;
        this.sexo = sexo;
    }
    private void validarDni(String dni) {
        if (dni == null || !dni.matches("^\\d{8}$")) {
            throw new DniInvalidoException("DNI inválido: debe contener exactamente 8 dígitos numéricos.");
        }
    }
    public String getDni() { 
        return this.dni; 
    }
    public String getNombres() { 
        return this.nombres; 
    }
    public String getApellidos() { 
        return this.apellidos; 
    }
    public LocalDate getFechaNacimiento() { 
        return this.fechaNacimiento; 
    }
    public String getSexo() { 
        return this.sexo; 
    }
    public String getNombreCompleto() { 
        return this.apellidos + ", " + this.nombres; 
    }
    public String getDniEnmascarado() {
        return "*****" + this.dni.substring(5);
    }
    public int calcularEdad() {
        return Period.between(this.fechaNacimiento, LocalDate.now()).getYears();
    }
    public void setNombres(String nombres) { 
        this.nombres = nombres; 
    }
    public void setApellidos(String apellidos) { 
        this.apellidos = apellidos; 
    }
    public void setFechaNacimiento(LocalDate fechaNacimiento) { 
        this.fechaNacimiento = fechaNacimiento; 
    }
    public void setSexo(String sexo) { 
        this.sexo = sexo; 
    }
    public abstract String obtenerRol();
}

File: python/modelo/persona.py
from abc import ABC, abstractmethod
from datetime import date
import re
from excepciones.errores_clinicos import DniInvalidoException
class Persona(ABC):
    def __init__(self, dni: str, nombres: str, apellidos: str, fecha_nacimiento: date, sexo: str):
        self.__validar_dni(dni)
        self.__dni = dni
        self.__nombres = nombres
        self.__apellidos = apellidos
        self.__fecha_nacimiento = fecha_nacimiento
        self.__sexo = sexo
    def __validar_dni(self, dni: str) -> None:
        if not re.match(r"^\d{8}$", str(dni)):
            raise DniInvalidoException(f"El DNI '{dni}' debe contener exactamente 8 dígitos numéricos.")
    @property
    def dni(self) -> str:
        return self.__dni
    @property
    def nombre_completo(self) -> str:
        return f"{self.__apellidos}, {self.__nombres}"
    def get_dni_enmascarado(self) -> str:
        return f"*****{self.__dni[-3:]}"
    def calcular_edad(self) -> int:
        hoy = date.today()
        edad = hoy.year - self.__fecha_nacimiento.year
        if (hoy.month, hoy.day) < (self.__fecha_nacimiento.month, self.__fecha_nacimiento.day):
            edad -= 1
        return edad
    @abstractmethod
    def obtener_rol(self) -> str:
        pass